agreement/proposal.go

```diff
@@ -189,9 +190,24 @@ func deriveNewSeed(address basics.Address, vrf *crypto.VRFSecrets, rnd round, pe
 	return
 }
 
+// verifyNewSeed checks the things in the header that can only be confirmed by
+// looking into the unauthenticatedProposal, namely the BlockSeed and the
+// Proposer.
 func verifyNewSeed(p unauthenticatedProposal, ledger LedgerReader) error {
 	value := p.value()
 	rnd := p.Round()
+
+	curParams := config.Consensus[p.BlockHeader.CurrentProtocol]
+	if curParams.Mining().Enabled {
+		if p.BlockHeader.Proposer != value.OriginalProposer {
+			return fmt.Errorf("payload has wrong proposer (%v != %v)", p.Proposer, value.OriginalProposer)
+		}
+	} else {
+		if !p.BlockHeader.Proposer.IsZero() {
```

Comment:
Is the right behavior to reject entirely or simply ignore the proposer/don't add to the block?

Reply:
This is a validator note deciding whether a block is valid and should be approved.  If people are tucking things into the proposer field before we turn on Mining, something is very weird and we should reject.

Now, certainly I could imagine a day where we want to keep the proposer field, but turn off mining. If we do that, we'll need separate consensus params.